lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,28 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+  #use a Hash to keep track of the number of duplicates
+  new_hash = Hash.new(0)
+  #iterate through each element in the array
+  arr.each do |idx|
+      #add all elements to the new_hash and add one to the key's value if there's a duplicate
+      new_hash[idx] += 1
+  end
+  #make sure that the Array passed in contained something
+  if new_hash.empty?
+      return nil
+  end
+  #----------------------------------------------------------------
+  #Do more iteration to check if one value is greater than the rest??? 
+  #----------------------------------------------------------------
+  #iterate through each key and value in new_hash
+  new_hash.each do |x, y|
+      #if the value is greater than the key then that key had duplicates in the original Array
+      if y > x
```

Comment:
You made a good point that I overlooked when writing out the syntax. I need to compare the values of the hash and not compare the values to their keys...right?